### DISCUSSION TITLE
Can't get WebSockets connection to establish in production.

### DISCUSSION THREAD

**Discussion by abaftiu** [django/channels#1965] [Help!]
I am looking for some help using Django Channels/WebSockets.

I am not able to make a connection in Production (only in localhost).

The error I am getting in Production is this:
```
[2023-01-17 09:41:57 +0000] [19615] [ERROR] Exception in ASGI application
OSError: Multiple exceptions: [Errno 111] Connection refused, [Errno 99] Cannot assign requested address
During handling of the above exception, another exception occurred:
redis.exceptions.ConnectionError: Error connecting to localhost:6379. Multiple exceptions: [Errno 111] Connection refused, [Errno 99] Cannot assign requested address.
```

Any help is highly appreciated.

**Comment by carltongibson** [comment]
Grrrr. I can't say anything about this. (It's not a Channels issue... — it's lower than that.) 

Top hit for "Cannot assign requested address" is: https://stackoverflow.com/questions/19246103/socket-errorerrno-99-cannot-assign-requested-address-and-namespace-in-python — that has the sort of detail I'd imagine you're looking for.
